### ISSUE TITLE
A more 'classic' port

### ISSUE THREAD

**Issue by aidancbrady** [calclavia/ICBM-Classic#4]
Hey @calclavia- I'm getting back into modding development and I'm going to try and actually do this port like I mentioned months ago.  I'm going to start out with a Forge-based port to 1.7.10, and if I have the time, migrate to the NOVA API afterwards.  Let me know if you want me to make it public.

**Comment by calclavia** [comment]
Sure. If you can, license it under LGPL v3.

**Comment by aidancbrady** [comment]
I've finished the port and uploaded it to this repository: https://github.com/aidancbrady/ICBM
It's loaded with bugs but I should be able to get it polished fairly quickly. I've made a few minor changes and refactored the mod to use my libraries instead of the outdated Resonant Engine. Feel free to add to it whenever you wish :)

**Comment by dmodoomsirius** [comment]
@aidancbrady Dark and i are working on that slowly. https://github.com/BuiltBrokenModding/ICBM-ClassicMissilePack
https://github.com/BuiltBrokenModding/ICBM

**Comment by DarkGuardsman** [comment]
@aidancbrady Only calclavia has permission to port, update, or maintain a different version of ICBM publicly. You can port a version for personal or educational purposes. However, I have to ask that you do not port a public version of ICBM. If you want to help you can contribute to the existing clasic port of ICBM that @dmodoomsirius linked.

Note I have full intellectual property rights to ICBM as part of my deal with Calclavia.

**Comment by aidancbrady** [comment]
@DarkGuardsman, I intend for this to be a continued version of the ICBM mod as it existed in 1.6.4, staying true to the original style. By doing so I have also stripped the mod of sentries and the reinforced building materials, as it seems you already are developing those in different mods. 

If you read above, Calclavia gave me permission to port and publicize the mod under LGPL v3. I would be happy to change the title of my port if you feel there is a conflict.

**Comment by dmodoomsirius** [comment]
@aidancbrady we already have a 1.7.10 version and we have the right to change icbm as we wish. If you want to help contribute to those repos fell free.

**Comment by aidancbrady** [comment]
You do have the right to change ICBM as you wish. Note I'm working on a separate project.

**Comment by dmodoomsirius** [comment]
Its under the same name as we already planned on doing a ICBM classic as i started above by linking the repo.

**Comment by aidancbrady** [comment]
Well in that case, I would be more than happy to change the name of my derived port. I already have plans to overhaul the assets in use, the mods should end up being practically indistinguishable if you decide to follow through.

**Comment by dmodoomsirius** [comment]
You are more then welcome to help us Aidan. We would welcome it. We have many mods to deal with as well. also if your on esper feel free to pm dmod.

**Comment by aidancbrady** [comment]
I'd be happy to pitch in if I have some spare time. I'm considering naming my port "MissileTech" to ensure our projects do not interfere :)

**Comment by dmodoomsirius** [comment]
thanks for understanding @aidancbrady

**Comment by DarkGuardsman** [comment]
@aidancbrady I don't just have permission to change it. I own the intellectual property rights and rights to the code. This is for both new versions and old version as part of my deal with calclavia. Where I gave up all my rights to Resonant Induction. Which included code from 5 other of my mods. This is why you can't release it under LGPL as it would violate the copy right. Calclavia doesn't seem to understand this which is why he keeps telling people they can port it. 

Now I can't stop you from making a mod containing missiles. However, you can not use any of the old code, or assets. As for part of copy right you can't make an exact copy of the concepts either. Which means if you want to do missiles. They need to be different then the existing concepts from ICBM.

Not trying to be mean, just informing you of copy right and trying to protect my ability to continue work on the mod. As you already own Mekanism which is a large populare mod. I only have ICBM at the moment and a handful of small mods. 

Now I have a github repo that is designed to be an exact copy of the old verison. Down to all the tech short of the sentry guns. It has a dev that should be working on it but you are welcome to contribute to that repo. Out side of this repo I can't allow you to make a direct port of ICBM. Even if it is under a different name as this would violate my copy right.

**Comment by calclavia** [comment]
My bad. @DarkGuardsman is correct. I got used to just saying "do whatever you want with my mods" and forgot that Dark had all rights.

**Comment by aidancbrady** [comment]
@DarkGuardsman, are you available to speak over IRC?  I feel like we could come to an agreement that would coincide with your terms and allow me to continue with my plans for the release of this mod.

**Comment by dmodoomsirius** [comment]
@aidancbrady are you on irc atm?

**Comment by DarkGuardsman** [comment]
@aidancbrady I've got a few mins before I need to get back to school work.

**Comment by DarkGuardsman** [comment]
Issue has been resolved in IRC chat. Agreement is @aidancbrady can make a port of the missiles in the old version of ICBM. So long as references are made to the old version on any download sites.

**Comment by ghost** [comment]
darkguard is wrong, lgpl3, not closed source, dg is wrong

what dg has turned it into isnt icbm, he fails

aidan can do what he wants, no forced references, no hitlering.

**Comment by DarkGuardsman** [comment]
Doesn't matter if its open sourced or not. I own the rights to the names, images, and assets for ICBM. Both in the current, future, and past versions. The nature of the code is not directly related to the nature of the rest of the project. 

Also, the code is no open source it was originally under @calclavia education public license. In transferring ownership to me I changed the license to my education public license. Which is basically the same thing with a more open-ended nature. 

As well @aidancbrady is still permitted to do the work he is doing. We resolved today's issue peacefully as it was a miss understanding. @aidancbrady agreed to change out the models to not look like a derivative of my work. I was never intending to do anything else than ensure our projects are not confused as the same mod. As each of us is working on different goals and plans.
